### ISSUE TITLE
Issues marked has having source maps when there are none

### ISSUE THREAD

**Issue by smeubank** [getsentry/sentry#49200]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. follow almost any platform guide for JS
2. add sentry to your application
3. send first error without uploading source maps
4. check the issue in sentry, you will see an issue with nothing but in app frames and no indication of missing stack trace to the untrained eye

<img width="857" alt="image" src="https://github.com/getsentry/sentry/assets/47563310/745f9a0f-d0f6-4f08-9046-6686f2556df8">

6. upload source maps
7. trigger error again you will see the stack trace properly source mapped

<img width="848" alt="image" src="https://github.com/getsentry/sentry/assets/47563310/c84f960c-e6ef-404d-a8a5-b6958e975a39">

### Expected Result

error events sent to a sentry project where no source maps have been uploaded should:
1. not be marked as having source maps
2. which will allow the product to behave accordingly and guide user to upload them

### Actual Result

Seemingly boat loads of issues are being marked as having source maps, when they most certainly do not

This is breaking product features to inform users that source maps are missing and why, as well throwing off analytics at a massive scale

### Product Area

Issues

### Link

https://steven-eubank.sentry.io/issues/4179268921/?query=is%3Aunresolved&referrer=issue-stream&stream_index=2

**Comment by smeubank** [comment]
moving info from slack:

> How does the ui detect if sourcemaps are missing?
Analytics: https://github.com/getsentry/sentry/blob/master/static/app/utils/events.tsx#L221-L231 
debugger: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/source_map_debug.py#L85-L93 which is scoped per-frame


> We always had `data.sourcemap` available. The reason why I used `raw_stacktrace` is that, as I thought, it's only available if sourcemaps were applied.
> And what was considered applied:
> - there is a valid event with an exception with at least 1 in-app frame
> - that frame points to a file we have access to
> - that file has a sourcemappingurl
> - we have access to sourcemap file
> 
> If all those points are met, we consider sourcemaps being applied, we produce `raw_stacktrace`, thus treat it as being processed
> 
> If any of the points above were not met, event never had `raw_stactrace` applied
> 
> Why it's invalidated right now, is that symbolicator *always* returns `raw_stacktrace`, even if no changes were performed; it'll be then just a copy of original ones
> 
> tl;dr for this now is, as Armin already mentioned, if `frame.data.sourcemap` is present, it means that we were able to read frame, fetch source, fetch sourcemap and proceed with processing. Any further errors are due to invalid sourcemap itself, eg. broken mapping or missing sourcecontents

**Comment by scefali** [comment]
I have a fix for this: https://github.com/getsentry/sentry/pull/49363 and https://github.com/getsentry/sentry/pull/49355
